Source/WebCore/style/StyleFontSizeFunctions.cpp

```diff
@@ -184,21 +184,32 @@ int legacyFontSizeForPixelSize(int pixelFontSize, bool shouldUseFixedDefaultSize
     return findNearestLegacyFontSize<float>(pixelFontSize, fontSizeFactors, mediumSize);
 }
 
-static float adjustedFontSize(float size, float sizeAdjust, float xHeight)
+static float adjustedFontSize(float size, float sizeAdjust, float metricValue)
 {
     if (!size)
         return 0;
 
-    float aspectValue = xHeight / size;
+    float aspectValue = metricValue / size;
     return size * (sizeAdjust / aspectValue);
 }
 
-float adjustedFontSize(float size, float sizeAdjust, const FontMetrics& metrics)
+float adjustedFontSize(float size, const FontSizeAdjust& sizeAdjust, const FontMetrics& metrics)
 {
-    if (!metrics.hasXHeight())
-        return size;
+    switch (sizeAdjust.metric) {
+    case FontSizeAdjust::Metric::CapHeight:
+        return metrics.hasCapHeight() ? adjustedFontSize(size, *sizeAdjust.value, metrics.floatCapHeight()) : size;
```

Comment:
I think we should add a link to https://github.com/w3c/csswg-drafts/issues/6384 saying that this behavior isn't well-defined in the spec.

Reply:
Added.